hs/src/Reach/Eval/Core.hs

```diff
@@ -1599,7 +1603,7 @@ evalPrim p sargs =
             (T_Array x_ty x_sz, T_Array y_ty y_sz) -> do
               meet_ty <- dtypeMeet (at, x_ty) (at, y_ty)
               let t = T_Array meet_ty (x_sz + y_sz)
-              let mkdv = DLVar at "array_concat" t
+              let mkdv = DLVar at Nothing t
```

Comment:
Isn't the idea to forget less stuff? Would it make sense to make this slot an `Either String Blah` instead of a `Maybe Blah` so that these informative strings can be preserved?

Reply:
In practice, these strings don't saying anything more than the srclocs, and I haven't even found them useful as the compiler author

Reply:
From what I can tell, this field was not used for anything/anywhere